chore: [IA-533] Zendesk permission detail screen

Undermaken opened the pull request:
## Short description
This PR fills the screen about the data shared with the assistance tool.

Note: IP address could be in V4 or V6 form
Note2: this PR also fix a warning since this trick
```typescript
customRightIcon={{
        iconName: "",
        onPress: () => true
      }}
```
causes this error
![Schermata 2021-12-07 alle 09 45 47](https://user-images.githubusercontent.com/822471/144996780-c4d9a760-dba6-49fb-9676-0a45aca2010c.png)

The fix consists to check if the icon is an empty string or not

### iOS emulator
https://user-images.githubusercontent.com/822471/145802246-a603bbe3-9ee7-4408-a9ac-e8ab8e67c1ac.mp4

debiff commented:
Wait until the eng translation is completed

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
       contactSupport: "Contact support"
       seeReports: "See reports in progress"
   askPermissions:
+    nameSurname: "Nome e Cognome"
```
```suggestion
    nameSurname: "Name and Surname"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
       seeReports: "See reports in progress"
   askPermissions:
+    nameSurname: "Nome e Cognome"
+    fiscalCode: "Codice fiscale"
```
```suggestion
    fiscalCode: "Fiscal code"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
   askPermissions:
+    nameSurname: "Nome e Cognome"
+    fiscalCode: "Codice fiscale"
+    emailAddress: "Indirizzo email"
```
```suggestion
    emailAddress: "Email address"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    nameSurname: "Nome e Cognome"
+    fiscalCode: "Codice fiscale"
+    emailAddress: "Indirizzo email"
+    deviceAndOS: "Dispositivo e sistema operativo"
```
```suggestion
    deviceAndOS: "Device and Operating System"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    fiscalCode: "Codice fiscale"
+    emailAddress: "Indirizzo email"
+    deviceAndOS: "Dispositivo e sistema operativo"
+    ipAddress: "Indirizzo IP"
```
```suggestion
    ipAddress: "IP address"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    emailAddress: "Indirizzo email"
+    deviceAndOS: "Dispositivo e sistema operativo"
+    ipAddress: "Indirizzo IP"
+    currentAppVersion: "Versione in uso dell'app"
```
```suggestion
    currentAppVersion: "Version of the app in use"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    ipAddress: "Indirizzo IP"
+    currentAppVersion: "Versione in uso dell'app"
+    identityProvider: "Identity Provider"
+    diagnosticData: "Dati di diagnostica"
```
```suggestion
    diagnosticData: "In-app navigation data"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    currentAppVersion: "Versione in uso dell'app"
+    identityProvider: "Identity Provider"
+    diagnosticData: "Dati di diagnostica"
+    diagnosticDataValue: "Dati non sensibili legati al funzionamento dell’app"
```
```suggestion
    diagnosticDataValue: "Such as the the last screen or message displayed in IO."
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    identityProvider: "Identity Provider"
+    diagnosticData: "Dati di diagnostica"
+    diagnosticDataValue: "Dati non sensibili legati al funzionamento dell’app"
+    stock: "Dati dell’avviso di pagamento"
```
```suggestion
    stock: "Payment Notice details"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    diagnosticData: "Dati di diagnostica"
+    diagnosticDataValue: "Dati non sensibili legati al funzionamento dell’app"
+    stock: "Dati dell’avviso di pagamento"
+    stockValue: "Incluso l’errore che hai appena ricevuto"
```
```suggestion
    stockValue: "Including the error you just received"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    diagnosticDataValue: "Dati non sensibili legati al funzionamento dell’app"
+    stock: "Dati dell’avviso di pagamento"
+    stockValue: "Incluso l’errore che hai appena ricevuto"
     header: "Contact support"
```
```suggestion
    header: "Write us"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    stock: "Dati dell’avviso di pagamento"
+    stockValue: "Incluso l’errore che hai appena ricevuto"
     header: "Contact support"
     title: "To continue, we need your consent"
```
```suggestion
    title: "Do you give us your consent?"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    stockValue: "Incluso l’errore che hai appena ricevuto"
     header: "Contact support"
     title: "To continue, we need your consent"
+    body: "Per ricevere supporto è necessario inviare al team di IO alcuni dati. Se neghi il consenso, potrai inviare una richiesta generica via email."
```
```suggestion
    body: "In order to respond to the support request, we need access to some personal data. If you deny consent, you’ll have to send us a generic request via e-mail."
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
     header: "Contact support"
     title: "To continue, we need your consent"
+    body: "Per ricevere supporto è necessario inviare al team di IO alcuni dati. Se neghi il consenso, potrai inviare una richiesta generica via email."
+    listHeader: "Verranno inviati i seguenti dati:"
```
```suggestion
    listHeader: "The following data will be sent:"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,8 +2901,21 @@ support:
+    listHeader: "Verranno inviati i seguenti dati:"
     cta:
       allow: "Allow"
       denies: "Do not allow"
```
```suggestion
      denies: "Deny"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,11 +2901,24 @@ support:
+    ipAddress: "IP address"
+    currentAppVersion: "Version of the app in use"
+    identityProvider: "Identity Provider"
+    diagnosticData: "In-app navigation data"
```
@debiff this is now called "In-app navigation data" (e.g.: the last screen you visited), so the string "diagnosticData" should be changed (See Figma for more details)

thisisjp commented on locales/it/index.yml:
```diff
@@ -2929,8 +2929,21 @@ support:
+    ipAddress: "Indirizzo IP"
+    currentAppVersion: "Versione in uso dell'app"
+    identityProvider: "Identity Provider"
+    diagnosticData: "Dati di diagnostica"
```
```suggestion
    diagnosticData: "Dati di navigazione in app IO"
```

thisisjp commented on locales/it/index.yml:
```diff
@@ -2929,8 +2929,21 @@ support:
+    currentAppVersion: "Versione in uso dell'app"
+    identityProvider: "Identity Provider"
+    diagnosticData: "Dati di diagnostica"
+    diagnosticDataValue: "Dati non sensibili legati al funzionamento dell’app"
```
```suggestion
    diagnosticDataValue: "Come ad esempio l’ultima schermata o messaggio visualizzato in app IO"
```

thisisjp commented on locales/en/index.yml:
```diff
@@ -2901,11 +2901,24 @@ support:
+    header: "Write us"
+    title: "Do you give us your consent?"
+    body: "In order to respond to the support request, we need access to some personal data. If you deny consent, you’ll have to send us a generic request via e-mail."
+    listHeader: "The following data will be sent:"
```
```suggestion
    listHeader: "If you consent, we'll access this data:"
```

Undermaken commented:
@fabriziofff could you check this commit?
https://github.com/pagopa/io-app/pull/3552/commits/86beec7c4064bb4afc5b9c8207aac6a3c5d45e1a
I changed the signature of `useIOSelector` 

With the previous type this statement was legit
```typescript
const isUserLoggedIn = useIOSelector(state =>
    isLoggedIn(state.someNotExistingProperty)
  );
```
since `state` (the state) was marked as `any`

Undermaken commented:
@thisisjp @debiff 
I updated this PR with the last changes done in the relative [Figma](https://www.figma.com/file/PsDwmBK5bqKZdXsycrTolb/Flusso-Assistenza-(Zendesk)?node-id=12%3A1798)

I updated also the video in the PR description

thisisjp commented:
@Undermaken I love the way you use TextEdit 🤣 In the 3rd scenario (guest user) there's a wrong row (payment-related info)

Undermaken commented:
> wrong

😄 
Don't care about it. The payment item won't show until we don't do this https://pagopa.atlassian.net/browse/IA-564
During the show, I forgot to switch off but actually it is not connected to the a real logic

fabriziofff commented:
> @fabriziofff could you check this commit? [86beec7](https://github.com/pagopa/io-app/commit/86beec7c4064bb4afc5b9c8207aac6a3c5d45e1a) I changed the signature of `useIOSelector`
> 
> With the previous type this statement was legit
> 
> ```ts
> const isUserLoggedIn = useIOSelector(state =>
>     isLoggedIn(state.someNotExistingProperty)
>   );
> ```
> 
> since `state` (the state) was marked as `any`

LGTM!

Undermaken commented on ts/store/reducers/authentication.ts:
```diff
@@ -176,7 +176,7 @@ function matchWithIdp<O>(
   whenWithIdp: (state: AuthenticationStateWithIdp) => O
 ): O {
-  if (state.kind === "LoggedOutWithoutIdp") {
+  if (state.kind === "LoggedOutWithoutIdp" || state.kind === undefined) {
```
Unfortunately this part is very weak. But your comment gives me an idea to improve:
The authentication store section is wiped when the user does logout
https://github.com/pagopa/io-app/blob/796dd83ec5b352edc6c2cf7b3c3abf2479ea56f8/ts/store/reducers/index.ts#L166
So that section could be undefined

I improved it by setting the initial state, so `state.kind` won't be `undefined`
here the commit 885b936768c6a969a4e0411cab670bee686de153

Undermaken commented on ts/features/zendesk/screens/ZendeskAskPermissions.tsx:
```diff
@@ -1,44 +1,217 @@
+import { getAppVersion } from "../../../utils/appVersion";
+import { zendeskSupportCancel } from "../store/actions";
 
+type Item = {
```
sure
added 4b10856e702a2cbb028b6fd284ba3429443d502b

Undermaken commented on ts/features/zendesk/screens/ZendeskAskPermissions.tsx:
```diff
@@ -1,44 +1,217 @@
+            {items.map((item, idx) => (
+              <ItemComponent key={`permission_item_${idx}`} {...item} />
+            ))}
+            <EdgeBorderComponent />
```
It is a mine idea 🙄 
I guess it could be removed
removed 8c949e40e0954e56add1a17246fb6dea4e685ddd

Undermaken commented on ts/features/zendesk/screens/ZendeskAskPermissions.tsx:
```diff
@@ -1,44 +1,217 @@
+  ];
+  const items = getItems(itemsProps)
+    .filter(it => (!assistanceForPayment ? it.id !== "paymentIssues" : true))
+    .filter(it => !itemsToRemove.includes(it.id ?? ""));
```
Good idea!
added 8c949e40e0954e56add1a17246fb6dea4e685ddd
